src/biotite/interface/rdkit/mol.py

```diff
@@ -237,14 +288,83 @@ def from_mol(mol, conformer_id=None, add_hydrogen=None):
         lambda: np.full(atoms.array_length(), "", dtype=object)
     )
     atoms.add_annotation("charge", int)
+    atoms.add_annotation("b_factor", float)
+    atoms.add_annotation("occupancy", float)
+    atoms.add_annotation("label_alt_id", str)
+
+    element_counter = defaultdict(lambda: 1)
     for rdkit_atom in rdkit_atoms:
+        _atom_idx = rdkit_atom.GetIdx()
+
+        # ... add standard annotations
+        element = rdkit_atom.GetSymbol().upper().strip()
+        atoms.element[_atom_idx] = element
+        atoms.charge[_atom_idx] = rdkit_atom.GetFormalCharge()
+
+        # ... add PDB related annotations
+        residue_info = rdkit_atom.GetPDBResidueInfo()
+        if residue_info is None:
+            residue_info = AtomPDBResidueInfo(
+                atomName="",
+                occupancy=0.0,
+                tempFactor=float("nan"),
+                altLoc=".",
+            )
+            if element == "H":
+                # ... attempt inferring residue information from nearest heavy atom
+                #     in case of a hydrogen atom without explicit residue information
+                nearest_heavy_atom = rdkit_atom.GetNeighbors()[0]
+                nearest_heavy_atom_res_info = nearest_heavy_atom.GetPDBResidueInfo()
+                if nearest_heavy_atom_res_info is not None:
+                    residue_info.SetChainId(nearest_heavy_atom_res_info.GetChainId())
+                    residue_info.SetResidueName(
+                        nearest_heavy_atom_res_info.GetResidueName()
+                    )
+                    residue_info.SetResidueNumber(
+                        nearest_heavy_atom_res_info.GetResidueNumber()
+                    )
+                    residue_info.SetInsertionCode(
+                        nearest_heavy_atom_res_info.GetInsertionCode()
+                    )
+                    residue_info.SetIsHeteroAtom(
+                        nearest_heavy_atom_res_info.GetIsHeteroAtom()
+                    )
+                    residue_info.SetAltLoc(nearest_heavy_atom_res_info.GetAltLoc())
+            element_count = element_counter[
+                (
+                    residue_info.GetChainId(),
+                    residue_info.GetResidueName(),
+                    residue_info.GetResidueNumber(),
+                    residue_info.GetInsertionCode(),
+                    element,
+                )
+            ]
+            residue_info.SetName(f"{element}{element_count}")
```

Comment:
I think we could omit naming atoms here and leave it up the user, like in `SDFile` where atom names cannot be loaded as well. There is even a convenience function for it: https://www.biotite-python.org/latest/apidoc/biotite.structure.create_atom_names.html

Reply:
Oh nice, I wasn't aware of that -- that sounds good to me, thanks for pointing out!